rosbridge_library/src/rosbridge_library/internal/subscribers.py

```diff
@@ -182,10 +183,12 @@ def subscribe(self, client_id, callback):
             # which adds the new callback to the subscriptions dictionary.
             self.new_subscriptions.update({client_id: callback})
             infos = self.node_handle.get_publishers_info_by_topic(self.topic)
-            if any(pub.qos_profile.durability == DurabilityPolicy.TRANSIENT_LOCAL for pub in infos):
+
+            if all(pub.qos_profile.durability == DurabilityPolicy.TRANSIENT_LOCAL for pub in infos):
```

Comment:
Probably need to also change it here:
```suggestion
            if len(infos) > 0 and all(
                pub.qos_profile.durability == DurabilityPolicy.TRANSIENT_LOCAL for pub in infos
            ):
```